### PR TITLE
feat: annual reports

### PR DESCRIPTION
**Pull request by gurjmatharu**
#458

### CODE REVIEW

File: app/cypress/integration/cif/project-revision/index.spec.js [hunk 1]
```diff
@@ -144,7 +161,7 @@ describe("the new project page", () => {
     });
   });
 
-  it("undoes changes on a new project when the user clicks the Undo Changes button", () => {
+  it.only("undoes changes on a new project when the user clicks the Undo Changes button", () => {
```
Comment (BCerki) [inline app/cypress/integration/cif/project-revision/index.spec.js:164]:
```suggestion
  it("undoes changes on a new project when the user clicks the Undo Changes button", () => {
```

File: app/tests/unit/pages/project-revision/[projectRevision]/form/annual-reports.test.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import { mocked } from "jest-mock";
+import { useRouter } from "next/router";
+import { getProjectRevisionPageRoute } from "pageRoutes";
+import { ProjectAnnualReportsPage } from "pages/cif/project-revision/[projectRevision]/form/annual-reports";
+import PageTestingHelper from "tests/helpers/pageTestingHelper";
+import compiledAnnualReportsFormQuery, {
+  annualReportsFormQuery,
+} from "__generated__/annualReportsFormQuery.graphql";
+
+jest.mock("next/dist/client/router");
```
Comment (BCerki) [inline app/tests/unit/pages/project-revision/[projectRevision]/form/annual-reports.test.tsx:12]: If https://github.com/bcgov/cas-cif/pull/605 goes in before this PR, there will be router-mocking helpers available

File: app/cypress/integration/cif/project-revision/index.spec.js [hunk 3]
```diff
@@ -125,16 +144,14 @@ describe("the new project page", () => {
       component: "Project Contacts Form",
       variant: "with errors",
     });
-    cy.findByText(/Quarterly reports/i).click();
-    cy.findByText(/Add quarterly reports/i).click();
-    cy.url().should("include", "/form/quarterly-reports");
 
-    cy.findByRole("button", { name: /add another quarterly report/i }).click();
-    cy.findByRole("button", { name: /add another quarterly report/i }).click();
-    cy.findByRole("button", { name: /add another quarterly report/i }).click();
```
Comment (BCerki) [inline app/cypress/integration/cif/project-revision/index.spec.js:134]: Why do we no longer need to check quarterly report validation?

**Review by BCerki — CHANGES_REQUESTED** [review]
I wasn't sure how to comment on a file that didn't have changes, so I'll just let you know here: the submit button in the quarterly report form should redirect to the new annual reports form rather than the summary page

File: app/cypress/integration/cif/project-revision/index.spec.js (hunk 3 above)
Comment (gurjmatharu) [reply app/cypress/integration/cif/project-revision/index.spec.js]: Mistake - readded.

File: app/components/Form/ProjectAnnualReportForm.tsx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+          formDataRecordId
+        }
+      }
+    `,
+    props.projectRevision
+  );
+  const [addAnnualReportMutation, isAdding] =
+    useAddReportingRequirementToRevision();
+
+  const addAnnualReport = (reportIndex: number) => {
```
Comment (dleard) [inline app/components/Form/ProjectAnnualReportForm.tsx:71]: Just a note: The next ~100 or so lines will be refactored in #636

File: schema/deploy/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 5]
```diff
@@ -0,0 +1,24 @@
+-- Deploy cif:computed_column/project_revision_annual_report_form_changes.sql to pg
+
+BEGIN;
```
Comment (dleard) [inline schema/deploy/computed_columns/project_revision_annual_report_form_changes.sql.sql:3]: lowercase

File: schema/deploy/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      and form_data_table_name='reporting_requirement'
+      and new_form_data->>'reportType'='Annual';
+
+$computed_column$ language sql stable;
+
+grant execute on function cif.project_revision_project_annual_report_form_changes to cif_internal, cif_external, cif_admin;
+
+comment on function cif.project_revision_project_annual_report_form_changes is 'Computed column for graphql to retrieve the changes related to the Annual reporting requirement records, within a project revision';
+
+COMMIT;
```
Comment (dleard) [inline schema/deploy/computed_columns/project_revision_annual_report_form_changes.sql.sql:24]: lowercase

File: schema/revert/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
+-- Revert cif:computed_column/project_revision_annual_report_form_changes.sql from pg
+
+BEGIN;
```
Comment (dleard) [inline schema/revert/computed_columns/project_revision_annual_report_form_changes.sql.sql:3]: lowercase

File: schema/revert/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+drop function cif.project_revision_project_annual_report_form_changes;
+
+COMMIT;
```
Comment (dleard) [inline schema/revert/computed_columns/project_revision_annual_report_form_changes.sql.sql:7]: lowercase

File: schema/verify/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 9]
```diff
@@ -0,0 +1,7 @@
+-- Verify cif:computed_column/project_revision_annual_report_form_changes.sql on pg
+
+BEGIN;
```
Comment (dleard) [inline schema/verify/computed_columns/project_revision_annual_report_form_changes.sql.sql:3]: lowercase

File: schema/verify/computed_columns/project_revision_annual_report_form_changes.sql.sql [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+
+select pg_get_functiondef('cif.project_revision_project_annual_report_form_changes(cif.project_revision)'::regprocedure);
+
+ROLLBACK;
```
Comment (dleard) [inline schema/verify/computed_columns/project_revision_annual_report_form_changes.sql.sql:7]: lowercase

**Review by dleard — CHANGES_REQUESTED** [review]

**Review by BCerki — APPROVED** [review]
Looks good to me!

**Review by dleard — APPROVED** [review]
:shipit:

**Pull request merged**